libs/common/include/launchdarkly/data/evaluation_detail.hpp

```diff
@@ -60,6 +60,14 @@ class EvaluationDetail {
      */
     [[nodiscard]] std::optional<EvaluationReason> const& Reason() const;
 
+    /**
+     * Check if an evaluation reason exists, and if so, if it is of a particular
+     * kind.
+     * @param kind Kind to check.
+     * @return True if a reason exists and matches the given kind.
+     */
+    [[nodiscard]] bool ReasonKindIs(enum EvaluationReason::Kind kind) const;
+
```

Comment:
I found myself needing to reach into the reason and check if it was a specific kind, like:
```c++
if (detail.Reason() && detail.Reason()->Kind() == EvaluationReason::Kind::kWhatever)
```
I feel like this could be a reasonably common task and deserves a method, but perhaps not.

Reply:
I suppose it doesn't hurt anything.